File: java-gradle/aimax/src/main/java/net/pijcke/nim/aimax/MaxAIFactory.java
package net.pijcke.nim.aimax;

import net.pijcke.nim.rules.Player;
import net.pijcke.nim.rules.PlayerFactory;

public class MaxAIFactory implements PlayerFactory {

    @Override
    public String label() {
        return "AI : Max";
    }

    @Override
    public Player build() {
        return new MaxAI();
    }
}

File: variants/kotlin-gradle/nim/aimax/src/main/kotlin/net/pijcke/nim/aimax/MaxAiFactory.kt
package net.pijcke.nim.aimax

import net.pijcke.nim.rules.Player
import net.pijcke.nim.rules.PlayerFactory

class MaxAiFactory : PlayerFactory {

    override fun label() = "AI : Max"

    override fun build(): Player {
        return MaxAi()
    }

}